### PR TITLE
YD-434 Embed Swagger UI

### PR DESCRIPTION
**Pull request by Bert-R**
By embedding Swagger UI, we are not longer dependent on the availability of http://petstore.swagger.io, nor do changes there immediately impact us. The Swagger spec was updated to make it work with the current version of Swagger UI. The most notable update is that the parameters are now defined as reusable objects. This makes the spec considerably shorter and it resolves the issue that somehow parameters defined for the path (rather than the operation) weren't recognized anymore.

With this update, it is possible to go to any installation of the Yona server and access Swagger UI on it: http://localhost:8082/swagger-ui/index.html

### CODE REVIEW

File: appservice/src/main/resources/swagger/swagger-spec.yaml [hunk 1]
```diff
@@ -1194,25 +928,13 @@ paths:
...
-        type: string
-      - name: Yona-Password
-        in: header
-        description: The device password
-        required: true
-        type: string
     get:
       summary: Get a message
       description: Fetches all messages of the user with the given ID
+      parameters:
```
Comment (aristotelos) [inline appservice/src/main/resources/swagger/swagger-spec.yaml]: Please add those parameters to the `delete` verb as well.

File: appservice/src/main/resources/swagger/swagger-spec.yaml [hunk 2]
```diff
@@ -43,21 +93,12 @@ paths:
...
-        required: true
-        type: string
     get:
       summary: Get a user
       description: Fetches the user identified by the given ID
       parameters:
-        - name: Yona-Password
-          in: header
-          description: The device password
-          required: false
```
Comment (aristotelos) [inline appservice/src/main/resources/swagger/swagger-spec.yaml]: This has changed to `required: true` now.

Comment (Bert-R) [inline appservice/src/main/resources/swagger/swagger-spec.yaml]: Done.

File: appservice/src/main/resources/swagger/swagger-spec.yaml (hunk 1 above)
Comment (Bert-R) [inline appservice/src/main/resources/swagger/swagger-spec.yaml]: Done

File: appservice/src/main/java/nu/yona/server/rest/SwaggerSpecController.java [hunk 3]
```diff
@@ -22,7 +23,7 @@
 import nu.yona.server.exceptions.YonaException;
 
 @Controller
-@RequestMapping(value = "/swagger/swagger-spec.yaml", produces = { "text/yaml" })
+@RequestMapping(value = "/swagger/swagger-spec.yaml", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/rest/SwaggerSpecController.java]: You could add a comment here.

File: appservice/src/main/resources/swagger/swagger-spec.yaml [hunk 4]
```diff
@@ -98,7 +98,11 @@ paths:
       description: Fetches the user identified by the given ID
       parameters:
         - $ref: "#/parameters/userID"
-        - $ref: "#/parameters/yonaPassword"
+        - name: Yona-Password
```
Comment (aristotelos) [inline appservice/src/main/resources/swagger/swagger-spec.yaml]: You could add a comment here why it is not required.

**Pull request merged**